llvm/lib/IR/Verifier.cpp

```diff
@@ -4285,6 +4291,12 @@ void Verifier::visitAllocaInst(AllocaInst &AI) {
   SmallPtrSet<Type*, 4> Visited;
   Check(AI.getAllocatedType()->isSized(&Visited),
         "Cannot allocate unsized type", &AI);
+  // Check if it's a target extension type that disallows being used on the
+  // stack.
+  if (auto *TTy = dyn_cast<TargetExtType>(AI.getAllocatedType())) {
+    Check(TTy->hasProperty(TargetExtType::CanBeLocal),
+          "Alloca has illegal target extension type", &AI);
```

Comment:
Also applies to the existing CanBeGlobal check, but this probably needs to be recursive? We'd not want to forbid `target("foo")` but then allow `{ target("foo") }`.

Reply:
Yeah, I asked about that here: https://discourse.llvm.org/t/rfc-adding-opaque-types-to-llvm-ir/65326/25

I might work on it but I want to keep that separate.

Reply:
#116639 does this for `CanBeGlobal`.

Reply:
I've now added the recursive checks to this patch too.